Signal/src/ViewControllers/ConversationView/ConversationViewItem.m

```diff
@@ -94,6 +93,9 @@ - (instancetype)initWithInteraction:(TSInteraction *)interaction
     _interaction = interaction;
     _isGroupThread = isGroupThread;
     _conversationStyle = conversationStyle;
+    _incomingMessageAuthorThread = ([interaction isKindOfClass:[TSIncomingMessage class]]
+            ? [TSContactThread getOrCreateThreadWithContactId:((TSIncomingMessage *)interaction).authorId]
```

Comment:
Now we're opening a sneaky read/write transaction. I equally don't think we should do this here or anywhere else on the main thread, especially for hot code like ConversationViewItem initialization.

Why do we now need a thread when we didn't before? I think it was part of collapsing the avatar factory methods behind a single unified method. Is that part of the change necessary?

Reply:
Keep in mind, we're already within the context of a read transaction here. 

Don't we already have access to a thread when ConversationViewItems are built? Could we just tag it on in the initializer?

Reply:
Oops, I was using the wrong method. Fixed.

>  I think it was part of collapsing the avatar factory methods behind a single unified method.

Nope, this is necessary for the new design.  If the conversation color for your 1:1 conversation with Alice is "red", Alice's default avatar should be red, even in group conversations.  This isn't something we were doing before, hence the need for the thread.

Reply:
Hmm, actually, this isn't good enough. I'm going to revisit tomorrow.

Reply:
> Nope, this is necessary for the new design. If the conversation color for your 1:1 conversation with Alice is "red", Alice's default avatar should be red, even in group conversations. This isn't something we were doing before, hence the need for the thread.

Oh right, that makes sense. 

Passing the thread in as an initializer param (e.g. instead of isGroupThread) seems reasonable.

Reply:
Okay, I just pushed a commit that addresses this issue.